### PR TITLE
[set] Add call to export set contents to an array.

### PR DESCRIPTION
**Pull request by andyjgf**

### CODE REVIEW

File: src/hb-bit-page.hh [hunk 1]
```diff
@@ -86,6 +86,58 @@ struct hb_bit_page_t
...
+  {
+    unsigned int count = 0;
+    for (unsigned i = 0; i < len () && count < size; i++)
+    {
+      elt_t bits = v[i];
+      uint32_t v_offset = i << ELT_BITS_LOG_2;
+      for (unsigned int j = 0; j < ELT_BITS && count < size; j++)
+	if ((elt_t (1) << j) & bits)
+	{
+	  *p++ = base | v_offset | j;
```
Comment (behdad) [inline src/hb-bit-page.hh:102]: Any reason you don't add `base` to `v_offset` outside the loop?

File: src/hb-set.cc [hunk 2]
```diff
@@ -239,6 +239,23 @@ hb_set_has (const hb_set_t *set,
...
+ * written out. Returns the number of codepoints written.
+ *
+ * Since: REPLACEME
+ */
+HB_EXTERN unsigned int
+hb_set_export_array (const hb_set_t *set,
+		      hb_codepoint_t *codepoints,
+		      unsigned int    size)
+{
+  if (unlikely(!set || !codepoints || !size)) return 0;
```
Comment (behdad) [inline src/hb-set.cc:254]: Remove these please.

File: src/hb-bit-page.hh (hunk 1 above)
Comment (andyjgf) [reply src/hb-bit-page.hh]: Good idea. Done.

File: src/hb-set.cc (hunk 2 above)
Comment (andyjgf) [reply src/hb-set.cc]: Done

**Pull request merged**